src/schema.js

```diff
@@ -258,6 +258,42 @@ const unflagPostSchema = Joi.object({
 			.required()
 	}
 })
+
+const updateUserSchema = Joi.object({
+	params: {
+		id: Joi.number()
+			.required()
+	},
+	body: {
+		firstName: Joi.string()
+			.alphanum()
+			.min(3)
+			.max(30)
+			.required(),
```

Comment:
I think this would be the significant review comment from me and it will change how some other parts are working now

the patch endpoint is meant to be a way to update any field in the user resource so none of the parameters needs to be required.

That is, if just the firstName is provided, we should update just the `firstName`.

Reply:
@jocrah I tried leaving all the fields as not required but it would try to update those fields with null values in the database

Reply:
yea understood @itzgeebee, but that's the major part of this task really. making sure any field that is not supplied in the request is not touched in the database

Kindly take a look at how we can do that. Would collaborate on it some time later if anything

Reply:
Alright

Reply:
@jocrah I have implemented a solution. kindly take a look

Reply:
cool, thanks.

its a nice fix. I think i see one or two scenarios where it might not work as expected but I will need some time to think through it and get back to you on it.

Reply:
Alright. Maybe you can tell me the scenarios too so I can also think on it @jocrah

Reply:
okay fair enough @itzgeebee 

i think i can summarise it into the fact that a user will not be able genuinely reset any of their optional fields due to the check we're doing

Ideally i'm more inclined to a fix that just updates the fields that came through no matter what they are (aside from the required ones of course). 

our current fix looks more like a whole resource update. I'm aware the ideal solution is not quite simple and i actually started looking into it before creating the issue hence me wanting a bit more time to think on it.

Reply:
@jocrah sure, I'm also looking at finding a solution that does a real partial update. I was just curious about the instances this solution might not work for or where an issue might arise.

Reply:
@itzgeebee so I just saw an approach where the update query is built based on the payload from the request

https://stackoverflow.com/a/21763631

seems okay for our purposes but do let me know if anything

Reply:
@jocrah Thanks. It was helpful

Reply:
Great